### PR TITLE
feat(stats): #MA-1034 rework of statistics count

### PR DESCRIPTION
**Pull request by BastienBenard**
## Describe your changes
Changed functions to count statistics

## Issue ticket number and link
[MA-1034](https://jira.support-ent.fr/browse/MA-1034)

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [x] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: statistics-presences/src/main/java/fr/openent/statistics_presences/indicator/IndicatorWorker.java [hunk 1]
```diff
@@ -35,13 +35,16 @@ public abstract class IndicatorWorker extends AbstractVerticle {
...
+        this.statisticsService = new DefaultStatisticsService("AllIndicator");
+    }
+
     @Override
     public void start() throws Exception {
         // for some reason we might lose our verticle's context and might also pick its parent's context to keep it "alive"
         // in order to avoid this behavior, we assign manually its own context to indicatorContext
         indicatorContext = vertx.getOrCreateContext();
         config = new JsonObject(config().toString());
-        this.statisticsService = new DefaultStatisticsService(this.indicatorName());
```
Comment (BastienBenard) [inline statistics-presences/src/main/java/fr/openent/statistics_presences/indicator/IndicatorWorker.java]: fix NPE

**Review by sjollois — APPROVED** [review]

File: statistics-presences/src/main/java/fr/openent/statistics_presences/indicator/IndicatorWorker.java [hunk 2]
```diff
@@ -35,13 +35,16 @@ public abstract class IndicatorWorker extends AbstractVerticle {
     protected Report report;
     protected StatisticsService statisticsService;
 
+    public IndicatorWorker() {
+        this.statisticsService = new DefaultStatisticsService("AllIndicator");
```
Comment (qpic-cgi) [inline statistics-presences/src/main/java/fr/openent/statistics_presences/indicator/IndicatorWorker.java]:
```suggestion
        this.statisticsService = new DefaultStatisticsService("Indicators");
```

**Review by qpic-cgi — APPROVED** [review]

File: statistics-presences/src/main/java/fr/openent/statistics_presences/bean/global/GlobalSearch.java [hunk 3]
```diff
@@ -435,7 +435,6 @@ private JsonObject match(List<String> types, boolean isTotalAbsences) {
 
         JsonObject matcher = new JsonObject()
                 .put("structure", this.filter.structure())
```
Comment (flmartineau) [inline statistics-presences/src/main/java/fr/openent/statistics_presences/bean/global/GlobalSearch.java]: Magic strings

File: statistics-presences/src/main/java/fr/openent/statistics_presences/bean/monthly/MonthlyGraphSearch.java [hunk 4]
```diff
@@ -119,7 +118,6 @@ private JsonObject match(List<String> types) {
 
         JsonObject matcher = new JsonObject()
                 .put("structure", this.filter.structure())
```
Comment (flmartineau) [inline statistics-presences/src/main/java/fr/openent/statistics_presences/bean/monthly/MonthlyGraphSearch.java]: Magic strings

File: statistics-presences/src/main/java/fr/openent/statistics_presences/bean/monthly/MonthlySearch.java [hunk 5]
```diff
@@ -180,7 +179,6 @@ private JsonObject match(List<String> types) {
 
         JsonObject matcher = new JsonObject()
                 .put("structure", this.filter.structure())
```
Comment (flmartineau) [inline statistics-presences/src/main/java/fr/openent/statistics_presences/bean/monthly/MonthlySearch.java]: Magic strings

File: statistics-presences/src/main/java/fr/openent/statistics_presences/indicator/IndicatorWorker.java (hunk 2 above)
Comment (flmartineau) [reply statistics-presences/src/main/java/fr/openent/statistics_presences/indicator/IndicatorWorker.java]: Field.INDICATORS même :)

**Review by flmartineau — APPROVED** [review]

**Pull request merged**